### PR TITLE
[HOPSWORKS-1424][featurestore] Handle upper/lower case for feature group names

### PR DESCRIPTION
**Pull request by moritzmeister**
https://logicalclocks.atlassian.net/browse/HOPSWORKS-1424

### CODE REVIEW

File: hops/tests/test_featurestore.py [hunk 1]
```diff
@@ -578,36 +578,37 @@ def test_do_prepare_stats_settings(self, sample_metadata):
         assert result["num_clusters"] == 10
         assert result["corr_method"] == "spearman"
 
-    def test_validate_metadata(self):
+    def test_validate_metadata(self, sample_metadata):
```
Comment (SirOibaf) [inline hops/tests/test_featurestore.py]: Could you please add a test to check the behaviour when passing capital letters names?

File: hops/featurestore_impl/util/fs_utils.py [hunk 2]
```diff
@@ -653,26 +653,28 @@ def _validate_metadata(name, dtypes, description):
...
+        :featurestore_regex: Regex string to match featuregroup/training dataset and feature names with
 
     Returns:
         None
 
     Raises:
         :ValueError: if the metadata does not match the required validation rules
     """
-    name_pattern = re.compile("^[a-zA-Z0-9_]+$")
+    name_pattern = re.compile(featurestore_settings.featurestore_regex)
```
Comment (SirOibaf) [inline hops/featurestore_impl/util/fs_utils.py]: You don't need to compile the regex every time you call the method.

File: hops/featurestore_impl/util/fs_utils.py [hunk 3]
```diff
@@ -653,26 +653,28 @@ def _validate_metadata(name, dtypes, description):
...
     if len(name) > 256 or name == "" or not name_pattern.match(name):
```
Comment (SirOibaf) [inline hops/featurestore_impl/util/fs_utils.py]: As for the other PR, you can avoid checking if the name is empty as it's already taken care by the regex.

File: hops/featurestore_impl/util/fs_utils.py [hunk 4]
```diff
@@ -653,26 +653,28 @@ def _validate_metadata(name, dtypes, description):
...
-                         " and must match the regular expression: ^[a-zA-Z0-9_]+$, the provided name: {} "
-                         "is not valid".format(name))
+        raise ValueError("Name of feature group/training dataset cannot be empty, cannot contain upper case characters,"
+                         " cannot exceed 256 characters, and must match the regular expression: {}, the provided name: "
+                         "{} is not valid".format(featurestore_settings.featurestore_regex, name))
     if len(dtypes) == 0:
         raise ValueError("Cannot create a feature group from an empty spark dataframe")
 
     for dtype in dtypes:
         if len(dtype[0]) > 767 or dtype[0] == "" or not name_pattern.match(dtype[0]):
```
Comment (SirOibaf) [inline hops/featurestore_impl/util/fs_utils.py]: Same as above

File: hops/featurestore_impl/util/fs_utils.py (hunk 2 above)
Comment (moritzmeister) [reply hops/featurestore_impl/util/fs_utils.py]: Moved compilation to settings

File: hops/featurestore_impl/util/fs_utils.py (hunk 3 above)
Comment (moritzmeister) [reply hops/featurestore_impl/util/fs_utils.py]: removed check for empty name

File: hops/featurestore_impl/util/fs_utils.py (hunk 4 above)
Comment (moritzmeister) [reply hops/featurestore_impl/util/fs_utils.py]: removed check for empty name as above.

File: hops/tests/test_featurestore.py (hunk 1 above)
Comment (moritzmeister) [reply hops/tests/test_featurestore.py]: Done. Added test for upper case feature group name and feature name, as well as empty names.

**Pull request merged**